File: OdontoFlowWeb/OdontFlow/src/app/core/model/product-view.model.ts
export interface ProductViewModel {
    id?: string;  
    name?: string;
    price?: Number;
    productCategory?: string;
    productCategoryId?: Number;
    priceFormatted?: string;
    isLabelRequired?: boolean;
    applyDiscount?: boolean;
    active?: boolean;
  }

  export interface CreateProductDTO {
    name?: string;
    productCategory?: Number;
    price?: string;
    isLabelRequired?: boolean;
    applyDiscount?: boolean;
    active?: boolean;
  }

  export interface UpdateProductDTO {
    id?: string;  
    name?: string;
    productCategory?: Number;
    price?: string;
    isLabelRequired?: boolean;
    applyDiscount?: boolean;
    active?: boolean;
  }

File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/ViewModel/Product/ProductViewModel.cs
namespace OdontFlow.Domain.ViewModel.Product;

public class ProductViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string ProductCategory { get; set; } = default!;
    public string PriceFormatted { get; set; } = default!;
    public decimal Price  { get; set; } = default!;
    public bool IsLabelRequired { get; set; }
    public bool ApplyDiscount { get; set; }
    public int ProductCategoryId { get; set; } = default!;
    public bool Active { get; set; }
}
